### PR TITLE
Controller/PreDemo3

### PR DESCRIPTION
**Pull request by alszla**
PDF to HTML conversion and PDF TO DOCX are now part of the conversion endpoint.
There were some refactoring to the ConvertPdfToHtml and ConvertHtmlToDoc classes.

### CODE REVIEW

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 1]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+            }
+        }
+
+        return new PPTtoPdfResponse(fileName, fileDownloadUri, outputChecksumString);
+    }
+
+    /**
+     * Converts from a PDF document to to a MS Word document based on a user-provided criteria: input, config and output.
+     *
+     * @param asset The input pdf document itself
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 2]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     * @param input A JSON formatted string (wo \n characters) that holds all the parameters tied to the input document
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 3]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     * @param config A JSON formatted string (wo \n characters) that holds all the parameters tied to the configuration
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:975]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 4]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     *               of the particular process of conversion.
+     * @param output A JSON formatted string (wo \n characters) that holds all the parameters tied to the output document
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 5]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
         return new PPTtoPdfResponse(fileName, fileDownloadUri, outputChecksumString);
     }
 
+    /**
+     * Converts from a pdf document to an html page based on a user-provided criteria: input, config and output.
+     *
+     * @param asset The input pdf file itself
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 6]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     * @param input A JSON formatted string (wo \n characters) that holds all the parameters tied to the input image
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 7]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     * @param config A JSON formatted string (wo \n characters) that holds all the parameters tied to the configuration
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:874]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 8]
```diff
@@ -860,6 +866,218 @@ else if (configJson.getString("metadata").equals("xmp")) {
...
+     *               of the particular process of conversion.
+     * @param output A JSON formatted string (wo \n characters) that holds all the parameters tied to the output page
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: At the end of the comment, you missed the point "."

File: src/main/java/com/fundation/webservice/model/ConvertHtmlToDoc.java [hunk 9]
```diff
@@ -27,8 +27,9 @@
      * @param criteriaConvert holds the source and destination file paths.
      */
     public void convert(CriteriaConvert criteriaConvert) {
+        CriteriaThumbnailImage criteria = (CriteriaThumbnailImage) criteriaConvert;
         try {
-            Process process = Runtime.getRuntime().exec("3rdparty/" + "pandoc/" + "pandoc.exe -o " + criteriaConvert.getDestPath() + " " + criteriaConvert.getSrcPath());
+            Process process = Runtime.getRuntime().exec(Directories.TOOLS_DIR.getDir() + Directories.PANDOC_DIR.getDir() + "pandoc.exe -o " + criteria.getDestPath() + criteria.getName() + criteria.getExt() + " " + criteria.getSrcPath());
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/ConvertHtmlToDoc.java]: In this line exceeds 120 lines in length please separate them for better code readability.

File: src/main/java/com/fundation/webservice/model/ConvertPdfToHtml.java [hunk 10]
```diff
@@ -27,8 +27,9 @@
      * @param criteriaConvert
      */
     public void convert(CriteriaConvert criteriaConvert) {
+        CriteriaThumbnailImage criteria = (CriteriaThumbnailImage) criteriaConvert;
         try {
-            Process process = Runtime.getRuntime().exec("3rdparty/" + "poppler/bin/" + "pdftohtml.exe -c -noframes " + criteriaConvert.getSrcPath() + " " + criteriaConvert.getDestPath());
+            Process process = Runtime.getRuntime().exec(Directories.TOOLS_DIR.getDir() + Directories.POPPLER_DIR.getDir() + "pdftohtml.exe -c -noframes " + criteria.getSrcPath() + " " + criteria.getDestPath() + criteria.getName() + criteria.getExt());
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/ConvertPdfToHtml.java]: In this line exceeds 120 lines in length please separate them for better code readability.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 1 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fixed.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 2 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fiexed

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 3 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:975]: Fixed

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 4 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fixed

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 5 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: fixed

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 6 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fixed.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 7 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:874]: Fixed.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 8 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fixed.

File: src/main/java/com/fundation/webservice/model/ConvertHtmlToDoc.java (hunk 9 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/ConvertHtmlToDoc.java]: Fixed.

File: src/main/java/com/fundation/webservice/model/ConvertPdfToHtml.java (hunk 10 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/ConvertPdfToHtml.java]: Fixed.

**Pull request merged**